crates/core/src/generated/ext.rs

```diff
@@ -32,6 +35,102 @@ impl From<bool> for Operand {
     }
 }
 
+impl TryFrom<Operand> for String {
+    type Error = ();
+
+    fn try_from(value: Operand) -> Result<Self, Self::Error> {
+        match value.value {
+            Some(operand::Value::StringValue(s)) => Ok(s),
+            _ => Err(()),
+        }
+    }
+}
+
+impl TryFrom<Operand> for f32 {
+    type Error = ();
+
+    fn try_from(value: Operand) -> Result<Self, Self::Error> {
+        match value.value {
+            Some(operand::Value::FloatValue(f)) => Ok(f),
+            _ => Err(()),
+        }
+    }
+}
+
+impl TryFrom<Operand> for usize {
+    type Error = ();
+
+    fn try_from(value: Operand) -> Result<Self, Self::Error> {
+        match value.value {
+            // [sic] TODO: we only have float operands, which is
+            // unpleasant. we should make 'int' operands a
+            // valid type, but doing that implies that the
+            // language differentiates between floats and
+            // ints, which it doesn't.
+            Some(operand::Value::FloatValue(f)) => Ok(f as usize),
```

Comment:
well this explains why I have a integer type - they want it, but don't have it ayy